nvflare/private/fed/client/client_app_runner.py

```diff
@@ -38,12 +37,14 @@ def __init__(self, time_out=60.0) -> None:
         self.timeout = time_out
         self.client_runner = None
 
-    def start_run(self, app_root, args, config_folder, federated_client, secure_train):
+    def start_run(self, app_root, args, config_folder, federated_client, secure_train, sp):
         self.client_runner = self.create_client_runner(app_root, args, config_folder, federated_client, secure_train)
+
         federated_client.set_client_runner(self.client_runner)
-        while federated_client.communicator.cell is None:
-            self.logger.info("Waiting for the client job cell to be created ....")
-            time.sleep(1.0)
+        federated_client.set_primary_sp(sp)
+
+        with self.client_runner.engine.new_context() as fl_ctx:
+            self.start_command_agent(args, federated_client, fl_ctx)
 
         self.sync_up_parents_process(federated_client)
```

Comment:
This call could raise exception when it fails to get clients info from server root. What's going to happen to the system if this exception occurs? Will the job cell be shutdown?

Reply:
If exception happens, the client job process will simply terminate. Error info in the log.